tools/flow9/backends/bprogram.flow

```diff
@@ -29,7 +29,7 @@ export {
 	// Statement-based languages can get this format:
 	BStatement ::= BIfStatement, BLetStatement, BSetMutableStatement, BReturnVoid, BReturn, BTailCall, BSequenceStatement, BSwitchStatement;
 		BIfStatement(cond : BExp, then : BStatement, else_ : BStatement);
-		BLetStatement(id : string, value : BExp, body : BStatement);
+		BLetStatement(id : string, value : BExp, body : BStatement, type : BType);
```

Comment:
Add a comment explaining that this is the type of the id, which can be a supertype of the the value. Do we also need a type for the body, in case that needs a supertype cast?